js/components/operatorSignIn/operatorSignInModal.tsx

```diff
@@ -138,6 +146,7 @@ const OperatorSignInModalContent = ({
           loading={loading}
           onComplete={() => {
             submit(badge, setComplete, setLoading);
+            onComplete();
```

Comment:
**question:** as I understand it, `submit` returns a promise but we're not actually waiting for that to resolve before calling `onComplete`, right? I worry that that might trigger a race condition where we might call `onComplete` and refresh the page but do so just quickly enough that it doesn't pick up the just-submitted data.

Reply:
One option would be to just make this callback another argument to the `submit` function which could then invoke it at the appropriate part of the lifecycle of the promise once the HTTP request has gone through.

Reply:
Ooh, good call out. I blanked that's in the promised land. fd3dcde